### PR TITLE
MM-20554 Migrate 'components/delete_post_modal' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by anchepiece**
#### Summary
This pull request migrates 'components/delete_post_modal' to TypeScript.
All tests pass locally, suggestions on improvement welcome.

#### Ticket Link
Fixes mattermost/mattermost-server#13687
JIRA https://mattermost.atlassian.net/browse/MM-20554

### CODE REVIEW

File: components/delete_post_modal/delete_post_modal.tsx [hunk 1]
```diff
@@ -1,32 +1,42 @@
...
+    };
+}
+
+type State = {
+    show: boolean;
+}
 
-    constructor(props) {
+export default class DeletePostModal extends React.PureComponent<Props, State> {
+    deletePostBtn: React.RefObject<HTMLButtonElement> = React.createRef();
```
Comment (devinbinnie) [inline components/delete_post_modal/delete_post_modal.tsx:37]: `React.createRef()` needs to happen in the constructor.

File: components/delete_post_modal/delete_post_modal.tsx [hunk 2]
```diff
@@ -90,7 +98,8 @@ export default class DeletePostModal extends React.PureComponent {
     }
 
     render() {
-        var commentWarning = '';
+        let commentWarning : React.ReactNode | string = '';
```
Comment (devinbinnie) [inline components/delete_post_modal/delete_post_modal.tsx:101]: `React.ReactNode` includes `string`, so you don't need to union this.

File: components/delete_post_modal/index.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import DeletePostModal from './delete_post_modal';
+
+type Actions = {
+    deleteAndRemovePost: (post: Post) => Promise<{data: boolean}>
+};
+
+function mapDispatchToProps(dispatch: Dispatch) {
+    return {
+
```
Comment (devinbinnie) [inline components/delete_post_modal/index.ts:20]: Nit: extra space

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks @anchepiece! Just a few comments to resolve.

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @anchepiece!

**Review by hmhealey — APPROVED** [review]
Thanks for helping with this migration. Looks good to me!

**Review by stevemudie — APPROVED** [review]
LGTM

**Pull request merged**